config/routes.rb

```diff
@@ -1,48 +1,60 @@
 Spree::Core::Engine.add_routes do
-  devise_for :spree_user,
-             :class_name => 'Spree::User',
-             :controllers => { :sessions => 'spree/user_sessions',
-                               :registrations => 'spree/user_registrations',
-                               :passwords => 'spree/user_passwords',
-                               :confirmations => 'spree/user_confirmations' },
-             :skip => [:unlocks, :omniauth_callbacks],
-             :path_names => { :sign_out => 'logout' },
-             :path_prefix => :user
+  if Spree::Auth::Engine.frontend_available?
+    devise_for(
+      :spree_user,
+      class_name: 'Spree::User',
+      controllers: {
+        sessions: 'spree/user_sessions',
+        registrations: 'spree/user_registrations',
+        passwords: 'spree/user_passwords',
+        confirmations: 'spree/user_confirmations'
+      },
+      skip: [:unlocks, :omniauth_callbacks],
+      path_names: { sign_out: 'logout' },
+      path_prefix: :user
+    )
```

Comment:
A small style nit - I'm never 100% sure of how to format these kinds of method calls (single-arg, then opts hash), but would prefer using an explicit hash like this:

```
devise_for :spree_user, {
  class_name: 'Spree::User',
  controllers: {
    sessions: 'spree/user_sessions',
    registrations: 'spree/user_registrations',
    passwords: 'spree/user_passwords',
    confirmations: 'spree/user_confirmations'
  },
  skip: [:unlocks, :omniauth_callbacks],
  path_names: { sign_out: 'logout' },
  path_prefix: :user
}
```